src/data-fetcher-loop/data-fetcher-loop.ts

```diff
@@ -73,10 +74,17 @@ export const runDataFetcher = async () => {
         .flat(2)
     );
 
+    // Compute the set of URLs coming from the config. These are trusted and don't need to be verified.
+    const trustedUrls = new Set(
+      Object.values(signedApiUrlsFromConfig)
+        .map((urlsPerProvider) => Object.values(urlsPerProvider))
+        .flat(2)
+    );
+
     // Better to log the non-decomposed object to see which URL comes from which chain-provider group.
-    logger.debug('Signed API URLs.', { signedApiUrls });
+    logger.debug('Signed API URLs.', { signedApiUrlsFromConfig, signedApiUrlsFromContract });
     const urls = uniq(
-      Object.values(signedApiUrls)
+      [...Object.values(signedApiUrlsFromConfig), ...Object.values(signedApiUrlsFromContract)]
```

Comment:
This is the key part of the PR - we now separate the config and contract specific URLs and merge them here.